components/lib/dataprotect/src/main/java/mozilla/components/lib/dataprotect/SecureAbove22Preferences.kt

```diff
@@ -59,9 +59,16 @@ private interface KeyValuePreferences {
  *
  * @param context A [Context], used for accessing [SharedPreferences].
  * @param name A name for this storage, used for isolating different instances of [SecureAbove22Preferences].
+ * @param useSecureIfAvailable A flag indicating whether or not to use encrypted storage when available.
+ * If set to `true`, will only by encrypted on Android API 23+. Otherwise, even if this flag is set to `true`,
+ * account state will be stored in plaintext.
  */
-class SecureAbove22Preferences(context: Context, name: String) : KeyValuePreferences {
-    private val impl = if (Build.VERSION.SDK_INT >= M) {
+class SecureAbove22Preferences(
+    context: Context,
+    name: String,
+    useSecureIfAvailable: Boolean = true
```

Comment:
I think something like `forceInsecure` is closer to your intent here, right?

An alternative would have been to just expose the `insecureimpl` class as public API, and let consumers choose it directly instead of going through the "compat" version.
I think I'd actually prefer that, it seems cleaner?